cmd/index.go

```diff
@@ -207,3 +220,113 @@ func restartSearchLogsFn(cmd *cobra.Command, args []string) {
 	log.Info("Success")
 	log.Infof("Total run time: %s", time.Now().Sub(clock).String())
 }
+
+func updateSynonymsFn(cmd *cobra.Command, args []string) {
+
+	clock := common.Init()
+	defer common.Shutdown()
+
+	folder, err := es.SynonymsFolder()
+	if err != nil {
+		log.Error(errors.Wrap(err, "SynonymsFolder not available."))
+		return
+	}
+	files, err := ioutil.ReadDir(folder)
+	if err != nil {
+		log.Error(errors.Wrap(err, "Cannot read synonym files list."))
+		return
+	}
+
+	bodyMask := `{
+		"index" : {
+			"analysis" : {
+				"filter" : {
+					"synonym_graph" : {
+						"type": "synonym_graph",
+						"tokenizer": "keyword",
+						"synonyms" : [
+							%s
+						]
+					}
+				}
+			}
+		}
+	}`
+
+	for _, fileInfo := range files {
+		keywords := make([]string, 0)
+
+		//  Convention: file name without extension is the language code.
+		var ext = filepath.Ext(fileInfo.Name())
+		var lang = fileInfo.Name()[0 : len(fileInfo.Name())-len(ext)]
+
+		indexName := es.IndexNameForServing("prod", consts.ES_RESULTS_INDEX, lang)
+
+		filePath := filepath.Join(folder, fileInfo.Name())
+		file, err := os.Open(filePath)
+		if err != nil {
+			log.Error(errors.Wrapf(err, "Unable to open synonyms file: %s.", filePath))
+			return
+		}
+		defer file.Close()
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			line := scanner.Text()
+
+			//  Blank lines and lines starting with pound are comments (like Solr format).
+			trimmed := strings.TrimSpace(line)
+			if trimmed != "" {
+				if !strings.HasPrefix(trimmed, "#") {
+					commaSeperated := strings.Replace(trimmed, "\t", ",", -1)
+					fline := fmt.Sprintf("\"%s\"", commaSeperated)
+					keywords = append(keywords, fline)
+				}
+			}
+		}
+
+		if err := scanner.Err(); err != nil {
+			log.Error(errors.Wrapf(err, "Error at scanning synonym config file: %s.", filePath))
+			return
+		}
+
+		synonymsBody := fmt.Sprintf(bodyMask, strings.Join(keywords, ","))
+
+		// Close the index in order to update the synonyms
+		closeRes, err := common.ESC.CloseIndex(indexName).Do(context.TODO())
+		if err != nil {
+			log.Error(errors.Wrapf(err, "CloseIndex: %s", indexName))
+			return
+		}
+		if !closeRes.Acknowledged {
+			log.Errorf("CloseIndex not Acknowledged: %s", indexName)
+			return
+		}
+
+		defer openIndex(indexName)
+		encodedIndexName := url.QueryEscape(indexName)
+		_, err = common.ESC.PerformRequest(context.TODO(), elastic.PerformRequestOptions{
```

Comment:
Why not use the API:
https://godoc.org/github.com/olivere/elastic#IndicesPutSettingsService
Please update.

Reply:
done